src/group-constructors.js

```diff
@@ -37,6 +41,61 @@
   };
 
   constructor.prototype = {
+    _isAncestor: function(effect) {
+      var a = this;
+      while (a !== null) {
+        if (a == effect)
+          return true;
+        a = a._parent;
+      }
+      return false;
+    },
+    _putChild: function(args, isAppend) {
+      var message = isAppend ? 'Cannot append an ancestor or self' : 'Cannot prepend an ancestor or self';
+      for (var i = 0; i < args.length; i++) {
+        if (this._isAncestor(args[i])) {
+          throw {
+            type: DOMException.HIERARCHY_REQUEST_ERR,
+            name: 'HierarchyRequestError',
+            message: message
+          };
+        }
+        isAppend ? this.children.push(args[i]) : this.children.unshift(args[i]);
+        args[i]._parent = this;
+      }
+      var node = this;
+      while (node) {
+        if (node.timing.duration === 'auto') {
+          node._timing.duration = node.activeDuration;
+        }
+        node = node._parent;
+      }
+      if (this.animation) {
+        var oldPlayState = this.animation.playState;
+        if (oldPlayState == 'pending') {
+          this.animation._needsRebuild = true;
+          scope.awaitStartTime(this.animation);
+        } else {
+          var oldCurrentTime = this.animation.currentTime;
+          var oldPlaybackRate = this.animation.playbackRate;
+          var effect = this.animation.effect;
+          this.animation.cancel();
+          this.animation = null;
+          document.timeline.play(effect);
+          this.animation.currentTime = oldCurrentTime;
```

Comment:
I think we should be setting the start time, not the current time. Otherwise the start time will still be unresolved.